OAK-9814: Improvements in NodeState/VersionCopier for visibility of p…

amit-jain opened the pull request:
…aths added and preserve sub-paths

- Added an optional consumer to be notified of any added paths
- Create a new preserveOnTarget which does not delete any sub-paths on target

Commit:
OAK-9814: Improvements in NodeState/VersionCopier for visibility of paths added and preserve sub-paths

- Added an optional consumer to be notified of any added paths
- Create a new preserveOnTarget which does not delete any sub-paths on target

amit-jain commented:
test failure in org.apache.jackrabbit.j2ee.TomcatIT due to timeouts and doesn't look related to the changes here

> java.lang.RuntimeException: java.net.SocketTimeoutException: Read timed out
> 	at com.gargoylesoftware.htmlunit.WebClient.download(WebClient.java:2378)

trekawek commented on oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/migration/NodeStateCopier.java:
```diff
@@ -87,18 +92,26 @@ public class NodeStateCopier {
-                            boolean referenceableFrozenNodes) {
+                            boolean referenceableFrozenNodes,
+                            boolean preserveOnTarget,
+                            Consumer<String> consumer) {
```
Let's call is `newNodesConsumer` or something like that.

trekawek reviewed (approved):
LGTM. Let's just make it clear what the new consumer accepts.

Commit:
OAK-9814: Improvements in NodeState/VersionCopier for visibility of paths added and preserve sub-paths

Review changes
- Rename consumer variable to make clear intention of consuming added paths only
